### PR TITLE
Adding authtoken provider type

### PR DESCRIPTION
**Pull request by sfeast**

### CODE REVIEW

**Comment by MeTaNoV** [comment]
seems legit!

**Comment by ebidel** [comment]
Can you point me to the docs for "authtoken" as a provider?

**Comment by MeTaNoV** [comment]
LMGTFY: https://www.firebase.com/docs/web/api/firebase/authwithoauthtoken.html :smile_cat:

**Comment by sfeast** [comment]
Yeah a little bit of a hack/confusing since it doesn't fit the existing provider switch statement pattern. "authtoken" isn't actually a provider, it's being used just to select the authwithoauthtoken authentication method in the switch statement, but a parameter with the desired provider type for authwithoauthtoken must be passed in as well, see https://github.com/sfeast/firebase-element-1/commit/176217f83237d2308630333a08f57ff39ddb8a29#diff-2eb10d852da705af9bafbcd8f6dc9d7dR231.

In other words, the difficulty was that the actual provider type here may also be google, facebook, twitter, etc, which conflicts with the existing use of these provider types in.

Hmm maybe better to do something similar to how "this.redirect" is used with these existing providers. i.e. add a new property like this.oauthtoken or similar to select authwithoauthtoken as the method to be used with the provider. Or something else?

**Comment by ebidel** [comment]
We merged a similar pr that covers this case a little cleaner.

https://github.com/GoogleWebComponents/firebase-element/pull/101/files

**Comment by sfeast** [comment]
Nice, looks great. Thanks.
